presto-main-base/src/main/java/com/facebook/presto/server/thrift/ColumnHandleThriftCodec.java

```
@@ -0,0 +1,82 @@
+ */
+package com.facebook.presto.server.thrift;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.drift.codec.CodecThriftType;
+import com.facebook.drift.codec.metadata.ThriftType;
+import com.facebook.drift.protocol.TProtocolReader;
+import com.facebook.drift.protocol.TProtocolWriter;
+import com.facebook.presto.connector.ConnectorCodecManager;
+import com.facebook.presto.metadata.HandleResolver;
+import com.facebook.presto.spi.ColumnHandle;
+
+import javax.inject.Inject;
+
+import java.nio.ByteBuffer;
+
+import static java.util.Objects.requireNonNull;
+
+public class ColumnHandleThriftCodec
+        extends AbstractTypedThriftCodec<ColumnHandle>
+{
+    private static final ThriftType THRIFT_TYPE = createThriftType(ColumnHandle.class);
+    private final ConnectorCodecManager connectorCodecManager;
+
+    @Inject
+    public ColumnHandleThriftCodec(HandleResolver handleResolver, ConnectorCodecManager connectorCodecManager, JsonCodec<ColumnHandle> jsonCodec)
+    {
+        super(ColumnHandle.class,
+                requireNonNull(jsonCodec, "jsonCodec is null"),
+                requireNonNull(handleResolver, "handleResolver is null")::getId,
+                handleResolver::getColumnHandleClass);
+        this.connectorCodecManager = requireNonNull(connectorCodecManager, "connectorThriftCodecManager is null");
+    }
+
+    @CodecThriftType
+    public static ThriftType getThriftType()
+    {
+        return THRIFT_TYPE;
+    }
+
+    @Override
+    public ThriftType getType()
+    {
+        return THRIFT_TYPE;
+    }
+
+    @Override
+    public ColumnHandle readConcreteValue(String connectorId, TProtocolReader reader)
+            throws Exception
+    {
+        ByteBuffer byteBuffer = reader.readBinary();
+        assert (byteBuffer.position() == 0);
+        byte[] bytes = byteBuffer.array();
+        return connectorCodecManager.getColumnHandleCodec(connectorId).map(codec -> codec.deserialize(bytes)).orElse(null);
```

I believe this should just throw to fail fast.

---

Added a throw